capirca/lib/paloaltofw.py

```diff
@@ -750,10 +752,27 @@ def __str__(self):
     rulebase = etree.SubElement(vsys_entry, "rulebase")
     security = etree.SubElement(rulebase, "security")
     rules = etree.SubElement(security, "rules")
+    tag = etree.Element("tag")
+
+    tag_num = 0
 
     # pytype: disable=key-error
     # pylint: disable=unused-variable
     for (header, pa_rules, filter_options) in self.pafw_policies:
+      tag_name = None
+      if header.comment:
+        comment = " ".join(header.comment).strip()
+        if comment:
+          tag_num += 1
+          # max tag len 127, max zone len 31
+          tag_name = self._TAG_NAME_FORMAT.format(
```

Comment:
Can we error out here if the tag name exceeds the max length? Or if the tag name is None.

Reply:
tag name len cannot exceed 127 unless zone(s) are > 31, and if zones are > 31 they are invalid.

I don't think format() can return None.

Reply:
Sounds good.